uat/testing-features/src/main/java/com/aws/greengrass/steps/MqttControlSteps.java

```diff
@@ -76,19 +79,34 @@ public void onMqttDisconnect(ConnectionControl connectionControl, Mqtt5Disconnec
 
     @Inject
     @SuppressWarnings("MissingJavadocMethod")
-    public MqttControlSteps(TestContext testContext, ScenarioContext scenarioContext, AWSResources resources,
-                            IotSteps iotSteps, EngineControl engineControl) throws IOException {
+    public MqttControlSteps(
+            TestContext testContext,
+            ScenarioContext scenarioContext,
+            AWSResources resources,
+            IotSteps iotSteps,
+            EngineControl engineControl,
+            GreengrassV2Client greengrassClient) throws IOException {
         this.testContext = testContext;
         this.scenarioContext = scenarioContext;
         this.resources = resources;
         this.iotSteps = iotSteps;
         this.engineControl = engineControl;
+        this.greengrassClient = greengrassClient;
         startMqttControl();
     }
 
-    @When("I associate {word} with ggc")
+    /**
+     * Associate client device with a core device .
+     *
+     * @param clientDeviceId string user defined client device id
+     */
+    @When("I associate {string} with ggc")
     public void associateClient(String clientDeviceId) {
-        //@TODO Implement method
+        final String coreName = testContext.coreThingName();
+        final String deviceName = testContext.testId()
+                                             .idFor(clientDeviceId);
+        greengrassClient.batchAssociateClientDeviceWithCoreDevice(b -> b.coreDeviceThingName(coreName)
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This code makes a call to a batch operation, but it does not handle any errors returned by that batch operation. Batch operations can succeed without throwing exceptions even if some items in the request fail. Consider checking the return value of the batch operation for failures.